lib/cyclopedio/mapping/wikipedia_category_utils.rb

```diff
@@ -0,0 +1,67 @@
+require 'cyclopedio/syntax'
+require 'wiktionary/noun'
+
+module Cyclopedio
+  module Mapping
+    # This class provides methods for processing category names and its parsed trees.
+    class WikipediaCategoryUtils
```

Comment:
OK. So the decorator class should be called NameDecorator. This decorator should not bother with Categor#parents and Category#children methods, since so far it is used with single category. You can move singularize_name to Nouns.